src/rgw/rgw_auth.h

```diff
@@ -39,6 +39,9 @@ class Identity {
   /* Return the ACLOwner for resources created by this identity. */
   virtual ACLOwner get_aclowner() const = 0;
 
+  /* Return the tenant of the ACLOwner for resources created by this identity. */
+  virtual std::string get_aclowner_tenant() const = 0;
```

Comment:
what's the difference between this and the existing `get_tenant()`?

Reply:
As I explained here (https://github.com/ceph/ceph/pull/59913#discussion_r1775777349) from what I understood, `get_tenant()` would not give us the _effective user's_ tenant. Am I mistaken?

Reply:
thanks for the reminder. @smanjara was recently looking into this for https://tracker.ceph.com/issues/69124, and i believe she was planning to override `SysReqApplier::get_tenant()` accordingly

Reply:
I guess that should be the same approach I would just need to integrate it to `SysReqApplier::get_tenant()` rather than introducing a new `get_aclowner_tenant()`. my motivation was to be able to let it distinguishable in case a `system_user` is also part of a tenant and that would be needed.

Reply:
I just integrated it with `get_tenant()` here https://github.com/ceph/ceph/pull/59913/commits/2c463d7755f5a05921209f3cfe786da527cb4290

Reply:
With the fact you brought here https://github.com/ceph/ceph/pull/59913#discussion_r1905863358 I guess for this PR I could make it simple without tackling https://tracker.ceph.com/issues/69124 (as that requires perhaps to consider both account and rgw_user case). So I just dropped those changes and now it just replies on `rgw_user`'s tenant field as you suggested.